### ISSUE TITLE
[1:33:35 PM] ERROR [worker reload] [worker init] /web/apps/web/.nuxt/dev/index.mjs failed

### ISSUE THREAD

**Issue by rocketiscool** [nuxt/nuxt#29882]
### Environment

- Operating System: `Darwin`
- Node Version:     `v20.14.0`
- Nuxt Version:     `3.14.159`
- CLI Version:      `3.15.0`
- Nitro Version:    `2.10.3`
- Package Manager:  `pnpm@9.12.3`
- Builder:          `-`
- User Config:      `default`
- Runtime Modules:  `nuxt-swiper@1.2.2`, `nuxt-gtag@3.0.1`, `@nuxtjs/seo@2.0.0-rc.23`, `@pinia/nuxt@0.7.0`, `@pinia-plugin-persistedstate/nuxt@1.2.1`, `nuxt-security@2.0.0`, `@vueuse/nuxt@11.2.0`, `nuxt-viewport@2.2.0`, `@sentry/nuxt/module@8.37.1`
- Build Modules:    `-`

### Reproduction

No reproduction neccessary

### Describe the bug

When I add '@sentry/nuxt/module' to my modules, I get this error

[worker reload] [worker init] ./nuxt/dev/index.mjs failed

 instantly after ℹ Vite client warmed up in 0ms

**Comment by rocketiscool** [comment]
When running pnpm run build I came across

```
ℹ Building Nuxt Nitro server (preset: node-server, compatibility date: 2024-11-04)            nitro 1:34:30 AM
[sentry-rollup-plugin] Info: Sending telemetry data on issues and performance to Sentry. To disable telemetry, set `options.telemetry` to `false`.

[nitro 1:35:18 AM]  ERROR  Error: ENOENT: no such file or directory, realpath '/Users/rocketiscool/Websites/site/web/node_modules/.pnpm/@resvg+resvg-js@2.6.2/node_modules/@resvg/resvg-js-android-arm64'


undefined


[1:35:18 AM]  ERROR  ENOENT: no such file or directory, realpath '/Users/rocketiscool/Websites/site/web/node_modules/.pnpm/@resvg+resvg-js@2.6.2/node_modules/@resvg/resvg-js-android-arm64
```

but the path definitely exists

**Comment by CptJJ** [comment]
Same here

**Comment by pi0** [comment]
Sharing a runnable reproduction would be a great help for us maintainers to pinpoint the issue and help you!

**Comment by CptJJ** [comment]
> Sharing a runnable reproduction would be a great help for us maintainers to pinpoint the issue and help you!

Sadly I have been struggling to recreate this myself. I have a content editor who works on windows run into this issue and been trying everything to get it resolved. Project runs fine for me and deploys fine also:

Downgrading nuxt did not fix like I hoped, aswell as trying other package managers (tried bun and pnpm) , as well as a full fresh clone and trying on a different windows laptop, 

 

![Image](https://github.com/user-attachments/assets/e9395379-c059-4ffe-a9a5-7ae154f9719f)

![Image](https://github.com/user-attachments/assets/ae8ec045-edfe-4189-a829-314bdbf85032)

also using @nuxtjs/seo@2.0.0-rc.23 and @vueuse/nuxt@11.2.0, nuxt seo has giving me some issues before in the past.

**Comment by rocketiscool** [comment]
Problem seems to be with something server related in the code, I have a lot of files so I can't pinpoint exactly where the issue is without removing files one by one which will cause the error of file not found instead.

I have 3 workspaces, two running nuxt and ones a layer which is extended in both.

This issue occurs only in one of my NUXT workspaces, the other NUXT workspace is perfectly fine and runs with sentry so it's very confusing and impossible to make a reproduction over.

**Comment by rdunk** [comment]
I'm seeing a similar issue in the Nuxt Sanity module.

Specifically, if I remove the `addServerHandler` calls [here](https://github.com/nuxt-modules/sanity/blob/main/src/module.ts#L350-L366), that specific error seems to disappear.

**Comment by derHodrig** [comment]
I have the same issue. I would assume it has something todo with nuxt/pinia v0.7.x

**Comment by falcon0ne** [comment]
For me it is same. I have two projects extended by a third project. One is running (also with Sentry) the other one not. I updated from Next 3.8.2 and now the issue appears in one of the projects. Unfortunately there is no "better" error message. With a clean fresh project, the issue does not appear, even with the same modules.

**Comment by derHodrig** [comment]
I got my app to work again.
In my case i got the same error but i dont have the same modules.
1. update your dependencies
2. try to build your app if possible `nuxt build`
this should hint you some errors
3. fix them
4. repeat step 2 until build successful
5. run dev server

The dev server was not able to show me type issues related with nuxt auto imports. (i really dont like the auto imports but many nuxt features rely on it; and for app.config.ts auto imports wont work)

So after I solved the type issues and i think the main issue of app.config.ts imports the app runs as expected...

My solution dont have to be your solution. The base problem is that the dev server breaks before it can log any issues.

**Comment by rocketiscool** [comment]
[nitro 3:41:07 PM]  ERROR  RollupError: server/db/index.ts (1:0): "PrismaClient" is not exported by "../../../database/src/generated/client/index.js", imported by "server/db/index.ts".


1: import { Prisma, PrismaClient } from '@rocketiscool/my-prisma-client'
   ^
2: import type { PaginationMeta } from '~~/../base/server/utils/createPaginationMeta'

Getting this when running pnpm run build to build NUXT server either workspace, only occurs when sentry is enabled as a module.
When sentry is disabled, pnpm run build works as should.

However, pnpm run dev still only causes the [ERROR [worker reload] [worker init] /web/apps/web/.nuxt/dev/index.mjs failed] on one NUXT workspace only

**Comment by pi0** [comment]
> Only occurs when sentry is enabled as a module.

@escrocket are you using the [beta SDK](https://docs.sentry.io/platforms/javascript/guides/nuxt/) of sentry?

**Comment by tobychidi** [comment]
> I got my app to work again. In my case i got the same error but i dont have the same modules.
> 
> 1. update your dependencies
> 2. try to build your app if possible `nuxt build`
>    this should hint you some errors
> 3. fix them
> 4. repeat step 2 until build successful
> 5. run dev server
> 
> The dev server was not able to show me type issues related with nuxt auto imports. (i really dont like the auto imports but many nuxt features rely on it; and for app.config.ts auto imports wont work)
> 
> So after I solved the type issues and i think the main issue of app.config.ts imports the app runs as expected...
> 
> My solution dont have to be your solution. The base problem is that the dev server breaks before it can log any issues.

This should be talked about more. When you run build you actually get to see the bugs causing the dev server to fail

**Comment by loama** [comment]
I've tried deleting the `.nuxt` folder, running `build`, `preview` everything works and doesn't have any issues.
But `dev` command still says `ERROR  [worker reload] [worker init] `

**Comment by derHodrig** [comment]
> I've tried deleting the `.nuxt` folder, running `build`, `preview` everything works and doesn't have any issues.
> But `dev` command still says `ERROR  [worker reload] [worker init] `

`nuxt typecheck` also okay?

**Comment by falcon0ne** [comment]
The Build was ok. For me it was a problem with sass. I deleted it manually and added the latest version again. And suddenly It worked.

**Comment by loama** [comment]
> > I've tried deleting the `.nuxt` folder, running `build`, `preview` everything works and doesn't have any issues.
> > But `dev` command still says `ERROR  [worker reload] [worker init] `
> 
> `nuxt typecheck` also okay?

Problem ended up being that I was using `runtimeConfig()` outside of a function. Which didn't trigger any warning/error outside of `dev`

**Comment by danielroe** [comment]
If there's not already an issue, might be worth creating one to track improving the error message here, upstream in https://github.com/unjs/nitro.

**Comment by NaviSparrow** [comment]
Maybe will help someone.
I had the same issue recently. I fixed it after searching for cjs require imports in project, (like a require "fs", require "path" etc). 
I found a few, replaced them for ESM imports, then error was gone, even with  latest Nuxt and Nitro (Nuxt 3.14.1592 with Nitro 2.10.4)

Also I noticed that the error is gone after removing the field "type: module" in package.json. But this was not an option for me. 
Error occurred only in dev mode, build was fine all along.
